app/cdap/components/PipelineDiff/util/__tests__/diff.tests.ts

```diff
@@ -0,0 +1,298 @@
+      const pipeline2 = {
+        stages: [exampleStage2, exampleStage1],
+        connections: [fromExample1ToExample2],
+      };
+      expect(computePipelineDiff(pipeline1, pipeline2).diffMap).toEqual({
+        connections: {},
+        stages: {},
+      });
+    });
+
+    it('should indicate no difference if pipelines are identical with different order of connections', () => {
+      const pipeline1 = {
+        stages: [exampleStage1, exampleStage2, exampleStage3],
+        connections: [fromExample1ToExample2, fromExample1ToExample3],
+      };
+      const pipeline2 = {
+        stages: [exampleStage1, exampleStage2, exampleStage3],
+        connections: [fromExample1ToExample3, fromExample1ToExample2],
+      };
+      expect(computePipelineDiff(pipeline1, pipeline2).diffMap).toEqual({
+        connections: {},
+        stages: {},
+      });
+    });
+
+    it('should indicate addition of a stage when a stage is added to pipeline1 to get pipeline2', () => {
+      const pipeline1 = {
+        stages: [exampleStage1, exampleStage2],
+        connections: [fromExample1ToExample2],
+      };
+      const pipeline2 = {
+        stages: [exampleStage1, exampleStage2, exampleStage3],
+        connections: [fromExample1ToExample2],
+      };
+      const pipeline3 = {
+        stages: [exampleStage1, exampleStage3, exampleStage2],
+        connections: [fromExample1ToExample2],
+      };
+
+      expect(computePipelineDiff(pipeline1, pipeline2).diffMap).toEqual({
+        connections: {},
+        stages: {
+          [exampleStage3DiffKey]: {
+            diff: exampleStage3,
+            stage1: undefined,
+            stage2: exampleStage3,
+            diffIndicator: DiffIndicator.ADDED,
+          },
+        },
+      });
+      expect(computePipelineDiff(pipeline1, pipeline3).diffMap).toEqual({
+        connections: {},
+        stages: {
+          [exampleStage3DiffKey]: {
+            diff: exampleStage3,
+            stage1: undefined,
+            stage2: exampleStage3,
+            diffIndicator: DiffIndicator.ADDED,
+          },
+        },
+      });
+    });
+
+    it('should indicate deletion of a stage when a stage is removed from pipeline1 to get pipeline2', () => {
+      const pipeline1 = {
+        stages: [exampleStage1, exampleStage2, exampleStage3],
+        connections: [fromExample1ToExample2],
+      };
+      const pipeline2 = {
+        stages: [exampleStage1, exampleStage2],
+        connections: [fromExample1ToExample2],
+      };
+
+      expect(computePipelineDiff(pipeline1, pipeline2).diffMap).toEqual({
+        connections: {},
+        stages: {
+          [exampleStage3DiffKey]: {
+            diff: exampleStage3,
+            stage1: exampleStage3,
+            stage2: undefined,
+            diffIndicator: DiffIndicator.DELETED,
+          },
+        },
+      });
+    });
+
+    it('should indicate modification of a stage when a stage from pipeline1 is modified to get pipeline2', () => {
```

Comment:
what's pipeline3, your description did not cover it

Reply:
`pipeline1` and `pipeline2` refers to the arguments of `computePipelineDiff` function. I updated the `describe` to indicate this and changed all constant names to `examplePipelineX`.